### ISSUE TITLE
Type Errors in 1.4.19

### ISSUE THREAD

**Issue by khunshan85** [Abhinandan-Kushwaha/react-native-gifted-charts#722]
```
components/molecules/DragAndDrop.tsx:133:7 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: AnySizeIProps | Readonly<AnySizeIProps>): AnySizeDragSortableView', gave the following error.
    Type '{ renderHeaderView?: any; headerViewHeight?: number | undefined; renderBottomView?: any; bottomViewHeight?: number | undefined; autoThrottle?: number | undefined; autoThrottleDuration?: number | undefined; ... 15 more ...; locale: Locale; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<AnySizeDragSortableView> & Readonly<AnySizeIProps>'.
      Property 'locale' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<AnySizeDragSortableView> & Readonly<AnySizeIProps>'.
  Overload 2 of 2, '(props: AnySizeIProps, context: any): AnySizeDragSortableView', gave the following error.
    Type '{ renderHeaderView?: any; headerViewHeight?: number | undefined; renderBottomView?: any; bottomViewHeight?: number | undefined; autoThrottle?: number | undefined; autoThrottleDuration?: number | undefined; ... 15 more ...; locale: Locale; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<AnySizeDragSortableView> & Readonly<AnySizeIProps>'.
      Property 'locale' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<AnySizeDragSortableView> & Readonly<AnySizeIProps>'.

133       locale={locale}
          ~~~~~~


node_modules/gifted-charts-core/src/components/BarAndLineChartsWrapper/index.ts:167:11 - error TS2802: Type 'IterableIterator<number>' can only be iterated through when using the '--downlevelIteration' flag or with a '--target' of 'es2015' or higher.

167     ? [...Array(noOfVerticalLines).keys()]
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/gifted-charts-core/src/components/BarAndLineChartsWrapper/index.ts:168:11 - error TS2802: Type 'IterableIterator<number>' can only be iterated through when using the '--downlevelIteration' flag or with a '--target' of 'es2015' or higher.

168     : [...Array(stackData ? stackData.length : data.length).keys()]
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/react-native-gifted-charts/src/index.tsx:9:3 - error TS1205: Re-exporting a type when 'isolatedModules' is enabled requires using 'export type'.

9   barDataItem,
    ~~~~~~~~~~~

node_modules/react-native-gifted-charts/src/index.tsx:10:3 - error TS1205: Re-exporting a type when 'isolatedModules' is enabled requires using 'export type'.

10   stackDataItem,
     ~~~~~~~~~~~~~

node_modules/react-native-gifted-charts/src/index.tsx:11:3 - error TS1205: Re-exporting a type when 'isolatedModules' is enabled requires using 'export type'.

11   BarChartPropsType,
     ~~~~~~~~~~~~~~~~~

node_modules/react-native-gifted-charts/src/index.tsx:12:3 - error TS1205: Re-exporting a type when 'isolatedModules' is enabled requires using 'export type'.

12   StackedBarChartPropsType,
     ~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/react-native-gifted-charts/src/index.tsx:13:3 - error TS1205: Re-exporting a type when 'isolatedModules' is enabled requires using 'export type'.

13   pieDataItem,
     ~~~~~~~~~~~

node_modules/react-native-gifted-charts/src/index.tsx:14:3 - error TS1205: Re-exporting a type when 'isolatedModules' is enabled requires using 'export type'.

14   PieChartPropsType,
     ~~~~~~~~~~~~~~~~~

node_modules/react-native-gifted-charts/src/index.tsx:15:3 - error TS1205: Re-exporting a type when 'isolatedModules' is enabled requires using 'export type'.

15   lineDataItem,
     ~~~~~~~~~~~~

node_modules/react-native-gifted-charts/src/index.tsx:16:3 - error TS1205: Re-exporting a type when 'isolatedModules' is enabled requires using 'export type'.

16   bicolorLineDataItem,
     ~~~~~~~~~~~~~~~~~~~

node_modules/react-native-gifted-charts/src/index.tsx:17:3 - error TS1205: Re-exporting a type when 'isolatedModules' is enabled requires using 'export type'.

17   LineChartPropsType,
     ~~~~~~~~~~~~~~~~~~

node_modules/react-native-gifted-charts/src/index.tsx:18:3 - error TS1205: Re-exporting a type when 'isolatedModules' is enabled requires using 'export type'.

18   LineChartBicolorPropsType,
     ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/react-native-gifted-charts/src/index.tsx:19:3 - error TS1205: Re-exporting a type when 'isolatedModules' is enabled requires using 'export type'.

19   popnPyramidDataItem,
     ~~~~~~~~~~~~~~~~~~~

node_modules/react-native-gifted-charts/src/index.tsx:20:3 - error TS1205: Re-exporting a type when 'isolatedModules' is enabled requires using 'export type'.

20   PopulationPyramidPropsType,
     ~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/react-native-gifted-charts/src/index.tsx:25:3 - error TS1205: Re-exporting a type when 'isolatedModules' is enabled requires using 'export type'.

25   DataSet,
     ~~~~~~~

node_modules/react-native-gifted-charts/src/LineChart/index.tsx:890:28 - error TS7053: Element implicitly has an 'any' type because expression of type 'number' can't be used to index type 'DataSet'.
  No index signature with a parameter of type 'number' was found on type 'DataSet'.

890         pointerItemLocal = set[pIndex];
                               ~~~~~~~~~~~


Found 17 errors in 4 files.

Errors  Files
     1  components/molecules/DragAndDrop.tsx:133
     2  node_modules/gifted-charts-core/src/components/BarAndLineChartsWrapper/index.ts:167
    13  node_modules/react-native-gifted-charts/src/index.tsx:9
     1  node_modules/react-native-gifted-charts/src/LineChart/index.tsx:890
error Command failed with exit code 2.
```

**Comment by Abhinandan-Kushwaha** [comment]
Hi @khunshan85 
`react-native-gifted-charts` is now shipping compiled js files instead of ts files from version `1.4.22` onwards. This should fix all type related issues. 🎉

**Comment by khunshan85** [comment]
Hi @Abhinandan-Kushwaha, I am now receiving the following errors:
`components/molecules/DragAndDrop.tsx:133:7 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: AnySizeIProps | Readonly<AnySizeIProps>): AnySizeDragSortableView', gave the following error.
    Type '{ renderHeaderView?: any; headerViewHeight?: number | undefined; renderBottomView?: any; bottomViewHeight?: number | undefined; autoThrottle?: number | undefined; autoThrottleDuration?: number | undefined; ... 15 more ...; locale: Locale; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<AnySizeDragSortableView> & Readonly<AnySizeIProps>'.
      Property 'locale' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<AnySizeDragSortableView> & Readonly<AnySizeIProps>'.
  Overload 2 of 2, '(props: AnySizeIProps, context: any): AnySizeDragSortableView', gave the following error.
    Type '{ renderHeaderView?: any; headerViewHeight?: number | undefined; renderBottomView?: any; bottomViewHeight?: number | undefined; autoThrottle?: number | undefined; autoThrottleDuration?: number | undefined; ... 15 more ...; locale: Locale; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<AnySizeDragSortableView> & Readonly<AnySizeIProps>'.
      Property 'locale' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<AnySizeDragSortableView> & Readonly<AnySizeIProps>'.

133       locale={locale}
          ~~~~~~


node_modules/gifted-charts-core/src/components/BarAndLineChartsWrapper/index.ts:172:11 - error TS2802: Type 'IterableIterator<number>' can only be iterated through when using the '--downlevelIteration' flag or with a '--target' of 'es2015' or higher.

172     ? [...Array(noOfVerticalLines).keys()]
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/gifted-charts-core/src/components/BarAndLineChartsWrapper/index.ts:173:11 - error TS2802: Type 'IterableIterator<number>' can only be iterated through when using the '--downlevelIteration' flag or with a '--target' of 'es2015' or higher.

173     : [...Array(stackData ? stackData.length : data.length).keys()]
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 3 errors in 2 files.

Errors  Files
     1  components/molecules/DragAndDrop.tsx:133
     2  node_modules/gifted-charts-core/src/components/BarAndLineChartsWrapper/index.ts:172`

**Comment by khunshan85** [comment]
Errors are fixed in the lates 1.4.23. Thanks @Abhinandan-Kushwaha
